lib/gemonames.rb

```diff
@@ -7,15 +7,34 @@ module Gemonames
   module_function
   BASE_API_URL = "http://api.geonames.org"
 
-  def client(username:, connection: nil)
-    connection ||= build_connection(username: username)
+  def client(username: nil, connection: nil, token: nil)
+    connection ||= if token
+                     build_connection_using_token(token: token)
+                   elsif username
+                     build_connection_using_username(username: username)
+                   else
+                     fail "username or token needs to be defined to build a client"
+                   end
+
     ApiClient.new(connection)
   end
 
-  def build_connection(username:)
+  def build_connection_using_username(username:)
+    build_connection do |faraday|
+      faraday.params[:username] = username
+    end
+  end
+
+  def build_connection_using_token(token:)
+    build_connection do |faraday|
+      faraday.params[:token] = token
+    end
+  end
```

Comment:
Those 2 methods look really similar, and they only differ by which hash key they set, maybe we could just do `faraday.params.merge` or `reverse_merge` and handle both cases in one method?